Detectors/TRD/base/include/TRDBase/Digit.h

```diff
@@ -32,11 +32,11 @@ using ArrayADC = std::array<ADC_t, constants::TIMEBINS>;
 // Digit class for TRD
 // Notes:
 //    Shared pads:
-//        the lowe mcm and rob is chosen for a given shared pad.
+//        the lower mcm and rob is chosen for a given shared pad.
 //        this negates the need for need alternate indexing strategies.
-//        it does however mean that if you are trying to go from pad/row to mcm/rob you need to remember to manually do the shared ones.
-//        TODO we could change the get methods to return the value in negaitve to indicate a shared pad, but I feel this is just added complexity? Comments?
-//        if you are going to have to check for negative you may as well check for it being shared.
+//        if you are trying to go from mcm/rob/adc to pad/row and back to mcm/rob/adc ,
+//        you may not end up in the same place, you need to remember to manually check for shared pads.
+
```

Comment:
I think here we should see what we do in the two use cases (I only see two):

 - in normal raw data processing, we get ROB/MCM/Ch and use this to construct the digit
 - for MC simulations, we get pad row/col, and we could have two digits corresponding the that pad. I see two options: we generate two digits, as for real data, or we need additional logic when we fill the MCM data structures for MC processing. I would suggest the first one.

Reply:
With reference to my comment at the bottom somewhere. you are defining a digit as a property of the MCM, which I would concur with.
The isshareddigit method can be trivially internalised into the class, and on requesting the pad/row which is shared, one returns both mcm/rob combinations. Technically how that works, I need to think about.
In the trap simulator for MC, the incoming digit data for pad/row is populated into both mcm for tracklet finding.

Reply:
As we discussed now, the TRAP simulator would need to be fixed to only read the corresponding digit for consistency. This might even have an impact on physics performance (although probably a tiny one.

I do not think there is a need to return the shared pad when retrieving the digits - and if so, then it would not be part of the `Digit` class, but the digit container (which is a list, so it does not support it anyways). So I think we don't need to do anything there, either.